src/Provider/FreeGeoIp/FreeGeoIp.php

```diff
@@ -83,4 +94,36 @@ public function getName(): string
     {
         return 'free_geo_ip';
     }
+
+    /**
+     * @param string $locale
+     */
+    public function setLocale(string $locale)
```

Comment:
I can understand that you feel it is pointless to have the `MessageFactory` as a second parameter. But that is a problem with the child classes but not the `AbstractHttpProvider`.

Feel free to open a PR about it. 

> And I think this violates the Liskov Substitution Principle.

No it does not, since the constructor is not a part of the `Provider` contract